### PR TITLE
[GH-19267] Spelling comments

### LINKED ISSUE #19267: Spelling errors in master branch
#### Summary
There are typos in mattermost-server

#### Steps to reproduce
How can we reproduce the issue (what version are you using?)
https://github.com/jsoref/mattermost-server/commit/4ff548f1235d2410473542bac8dd6262488c3561#commitcomment-62819286

#### Expected behavior

No typos and a ✅ 

#### Observed behavior (that appears unintentional) 
What did you see happen? Please include relevant error messages and/or screenshots.
❌ 
<details><summary>Unrecognized words (140)</summary>

```
activitated
activiy
acually
adjustement
alfter
allways
Ancilary
attachements
attachemtns
Avaliable
brakets
cannonical
categorywhich
cateory
catetories
cellspcing
Chaned
chaneld
Chanenel
Channe
comforms
commmandline
Complicance
constained
coresponding
corrensponding
couldnt't
Couns
cxt
deacivated
deactive
decprecated
depreacting
deriviative
destionation
deteminism
diemsnions
direferen
documentatin
duplcates
elipsis
emai
Emojies
emptively
Enivironment
enteprise
exaple
Exeeded
exepected
Exipry
exmaple
Failied
floag
forign
Fowarded
htp
idenitify
indentify
independant
initializaing
Initializiation
initialzed
innapropriate
invate
ivalid
Managment
markteplace
mattemost
memebership
mermbership
Metadat
metion
multipile
notexist
notifed
notoverriden
occured
ocurrs
oprphaned
orphanded
overrideable
overriden
overrite
overwites
pagnation
panicing
paramater
Patchs
Pemissions
permenent
permisssions
persistant
pontential
Posible
Predictible
prepackged
previoous
privat
promete
qouted
Receipients
recorgnizing
registeration
repoy
requrie
Resctricted
resetted
Respose
Restricions
retranslateed
retreiving
Roundtripper
Rountrippers
schemd
Schme
Searc
serch
shoudn't
should'nt
sigantures
signatrue
singup
somthing
specfified
STMP
Sucess
Suggesitons
Suscription
Sycn
syncronize
syncronizer
Sytem
theoritcally
thumbail
unknwon
updation
Usersname
webhok
Webhoook
wilrdcar
```
</details>

#### Possible fixes

- [ ] #19268
- [ ] strings
- [ ] struct fields and others

### PR DESCRIPTION
**Pull request by jsoref**
#### Summary

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/mattermost-server/commit/4ff548f1235d2410473542bac8dd6262488c3561#commitcomment-62819286

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/mattermost-server/commit/0c022a90db60cfacffe43b1986153121fe95c5fd

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I intend to annotate this PR shortly. I'm happy to drop changes to files, drop specific terms, or split things as requested, although this may take some time.

#### Ticket Link

Fixes #19267 

#### Release Note

```release-note

```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
           steps:
             # Taken from https://github.com/entur/owasp-orb/blob/master/src/%40orb.yml#L349-L361
             - owasp/generate_cache_keys:
-                cache_key: commmandline-default-cache-key-v7
+                cache_key: commandline-default-cache-key-v7
```
Comment (jsoref) [inline .circleci/config.yml:81]: Relatively harmless?

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
+.PHONY: build package run stop run-client run-server run-haserver stop-haserver stop-client stop-server restart restart-server restart-client restart-haserver start-docker clean-dist clean nuke check-style check-client-style check-server-style check-unit-tests test dist prepare-enterprise run-client-tests setup-run-client-tests cleanup-run-client-tests test-client build-linux build-osx build-windows package-prep package-linux package-osx package-windows internal-test-web-client vet run-server-for-web-client-tests diff-config prepackaged-plugins prepackaged-binaries test-server test-server-ee test-server-quick test-server-race start-docker-check migrations-bindata new-migration migration-prereqs
```
Comment (jsoref) [inline Makefile:1]: This target doesn't appear to exist...

File: Makefile [hunk 3]
```diff
@@ -47,7 +47,7 @@ else
 endif
 BUILD_WEBAPP_DIR ?= ../mattermost-webapp
 BUILD_CLIENT = false
-BUILD_HASH_CLIENT = independant
+BUILD_HASH_CLIENT = independent
```
Comment (jsoref) [inline Makefile:50]: ?

File: api4/apitestlib.go [hunk 4]
```diff
@@ -614,7 +614,7 @@ func (th *TestHelper) SetupSamlConfig() {
 		*cfg.SamlSettings.Enable = true
 		*cfg.SamlSettings.Verify = false
 		*cfg.SamlSettings.Encrypt = false
-		*cfg.SamlSettings.IdpURL = "https://does.notmatter.com"
+		*cfg.SamlSettings.IdpURL = "https://does.notmatter.example"
```
Comment (jsoref) [inline api4/apitestlib.go:617]: https://datatracker.ietf.org/doc/html/rfc6761#section-6.5

File: api4/cloud.go [hunk 5]
```diff
@@ -370,7 +370,7 @@ func getSubscriptionInvoicePDF(c *Context, w http.ResponseWriter, r *http.Reques
 
 	pdfData, filename, appErr := c.App.Cloud().GetInvoicePDF(c.AppContext.Session().UserId, c.Params.InvoiceId)
 	if appErr != nil {
-		c.Err = model.NewAppError("Api4.getSuscriptionInvoicePDF", "api.cloud.request_error", nil, appErr.Error(), http.StatusInternalServerError)
+		c.Err = model.NewAppError("Api4.getSubscriptionInvoicePDF", "api.cloud.request_error", nil, appErr.Error(), http.StatusInternalServerError)
```
Comment (jsoref) [inline api4/cloud.go:373]: ?

File: api4/team.go [hunk 6]
```diff
@@ -1377,12 +1377,12 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 func inviteGuestsToChannels(c *Context, w http.ResponseWriter, r *http.Request) {
 	graceful := r.URL.Query().Get("graceful") != ""
 	if c.App.Srv().License() == nil {
-		c.Err = model.NewAppError("Api4.InviteGuestsToChannels", "api.team.invate_guests_to_channels.license.error", nil, "", http.StatusNotImplemented)
+		c.Err = model.NewAppError("Api4.InviteGuestsToChannels", "api.team.invite_guests_to_channels.license.error", nil, "", http.StatusNotImplemented)
```
Comment (jsoref) [inline api4/team.go:1380]: ?

File: api4/team.go [hunk 7]
```diff
@@ -1377,12 +1377,12 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 
 	if !*c.App.Config().GuestAccountsSettings.Enable {
-		c.Err = model.NewAppError("Api4.InviteGuestsToChannels", "api.team.invate_guests_to_channels.disabled.error", nil, "", http.StatusNotImplemented)
+		c.Err = model.NewAppError("Api4.InviteGuestsToChannels", "api.team.invite_guests_to_channels.disabled.error", nil, "", http.StatusNotImplemented)
```
Comment (jsoref) [inline api4/team.go:1385]: ?

File: app/bot.go [hunk 8]
```diff
@@ -366,7 +366,7 @@ func (a *App) PermanentDeleteBot(botUserId string) *model.AppError {
 		var invErr *store.ErrInvalidInput
 		switch {
 		case errors.As(err, &invErr):
-			return model.NewAppError("PermanentDeleteBot", "app.bot.permenent_delete.bad_id", map[string]interface{}{"user_id": invErr.Value}, invErr.Error(), http.StatusBadRequest)
+			return model.NewAppError("PermanentDeleteBot", "app.bot.permanent_delete.bad_id", map[string]interface{}{"user_id": invErr.Value}, invErr.Error(), http.StatusBadRequest)
```
Comment (jsoref) [inline app/bot.go:369]: It's possible these are ui facing strings in which case a number of these items would have an l10n impact...

Generally, while localizers can treat ids as opaque, it's better for them not to have typos...

File: app/channel_test.go [hunk 9]
```diff
@@ -2143,7 +2143,7 @@ func TestMarkChannelAsUnreadFromPostCollapsedThreadsTurnedOff(t *testing.T) {
 	// user2: first root mention @user1
 	//   - user1: hello
 	//   - user2: mention @u1
-	//   - user1: another repoy
+	//   - user1: another reply
```
Comment (jsoref) [inline app/channel_test.go:2146]: ?

File: app/compliance.go [hunk 10]
```diff
@@ -65,7 +65,7 @@ func (a *App) GetComplianceReport(reportId string) (*model.Compliance, *model.Ap
 		var nfErr *store.ErrNotFound
 		switch {
 		case errors.As(err, &nfErr):
-			return nil, model.NewAppError("GetComplicanceReport", "app.compliance.get.finding.app_error", nil, nfErr.Error(), http.StatusNotFound)
+			return nil, model.NewAppError("GetComplianceReport", "app.compliance.get.finding.app_error", nil, nfErr.Error(), http.StatusNotFound)
 		default:
 			return nil, model.NewAppError("GetComplianceReport", "app.compliance.get.finding.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (jsoref) [inline app/compliance.go:70]: Seems like a good change...

File: app/remote_cluster_test.go [hunk 11]
```diff
@@ -41,7 +41,7 @@ func TestAddRemoteCluster(t *testing.T) {
 		remoteCluster := &model.RemoteCluster{
 			RemoteTeamId: model.NewId(),
 			Name:         "test2",
-			SiteURL:      "http://www2.exmaple.com:8065",
+			SiteURL:      "http://www2.example.com:8065",
```
Comment (jsoref) [inline app/remote_cluster_test.go:44]: If you need a second domain, you can use `example.net`...

File: app/user_test.go [hunk 12]
```diff
@@ -524,7 +524,7 @@ func TestUpdateUserEmail(t *testing.T) {
 			tokens, err = th.App.Srv().Store.Token().GetAllTokensByType(TokenTypeVerifyEmail)
 			// We verify the same conditions as the earlier function,
 			// but we also need to ensure that this is not the same token
-			// as before, which is possible if the token updation goroutine
+			// as before, which is possible if the token update goroutine
```
Comment (jsoref) [inline app/user_test.go:527]: ?

File: app/webhook_test.go [hunk 13]
```diff
@@ -646,8 +646,8 @@ func TestTriggerOutGoingWebhookWithUsernameAndIconURL(t *testing.T) {
 				EnablePostUsernameOverride: true,
 				EnablePostIconOverride:     true,
 				ExpectedUsername:           "webhookuser",
-				ExpectedIconURL:            "http://webhok/icon",
-				WebhookResponse:            &model.OutgoingWebhookResponse{Text: &webHookResponse, Username: "webhookuser", IconURL: "http://webhok/icon"},
+				ExpectedIconURL:            "http://webhook/icon",
```
Comment (jsoref) [inline app/webhook_test.go:649]: ?

File: config/common_test.go [hunk 14]
```diff
@@ -147,7 +147,7 @@ func TestConfigEnvironmentOverrides(t *testing.T) {
 	base, err := NewStoreFromBacking(memstore, nil, false)
 	require.NoError(t, err)
 	originalConfig := &model.Config{}
-	originalConfig.ServiceSettings.SiteURL = model.NewString("http://notoverriden.ca")
+	originalConfig.ServiceSettings.SiteURL = model.NewString("http://notoverridden.ca")
 
 	os.Setenv("MM_SERVICESETTINGS_SITEURL", "http://overridden.ca")
```
Comment (jsoref) [inline config/common_test.go:152]: This is bad and should live in an example domain or in a mattermost domain...

File: i18n/en.json [hunk 15]
```diff
@@ -2090,15 +2090,15 @@
     "translation": "Error getting access token from DB before deletion."
   },
   {
-    "id": "api.oauth.singup_with_oauth.disabled.app_error",
+    "id": "api.oauth.signup_with_oauth.disabled.app_error",
```
Comment (jsoref) [inline i18n/en.json:2093]: These keys obviously have localization fallout

File: i18n/en.json [hunk 16]
```diff
@@ -3121,7 +3121,7 @@
   },
   {
     "id": "api.templates.cloud_welcome_email.add_apps_sub_info",
-    "translation": "Streamline your work with tools like Github, Jira and Zoom. Explore all of the integrations we have on our"
+    "translation": "Streamline your work with tools like GitHub, Jira and Zoom. Explore all of the integrations we have on our"
```
Comment (jsoref) [inline i18n/en.json:3124]: Brand

File: i18n/en_AU.json [hunk 17]
```diff
@@ -3089,7 +3089,7 @@
   },
   {
     "id": "ent.migration.migratetoldap.duplicate_field",
-    "translation": "Unable to migrate AD/LDAP users with specified field. Duplicate entry detected. Please remove all duplcates and try again."
+    "translation": "Unable to migrate AD/LDAP users with specified field. Duplicate entry detected. Please remove all duplicates and try again."
```
Comment (jsoref) [inline i18n/en_AU.json:3092]: Sometimes projects don't want localizations to be touched. It's trivial for me to drop these changes if requested, or to apply the same changes to the other locales as directed.

File: model/search_params_test.go [hunk 18]
```diff
@@ -393,7 +393,7 @@ func TestParseSearchFlags2(t *testing.T) {
 			},
 		},
 		{
-			Name:  "string with a non-floag followed by :",
+			Name:  "string with a non-flag followed by :",
```
Comment (jsoref) [inline model/search_params_test.go:396]: ?

File: model/utils_test.go [hunk 19]
```diff
@@ -885,7 +885,7 @@ func TestIsValidHTTPURL(t *testing.T) {
 		},
 		{
 			"url with invalid scheme",
-			"htp://mattermost.com",
+			"http://mattermost.com",
```
Comment (jsoref) [inline model/utils_test.go:888]: ok, this fix is wrong. But, this isn't a reasonable scheme. Use `example:`

https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml

`htp` could be registered.

File: services/docextractor/docextractor_test.go [hunk 20]
```diff
@@ -180,7 +180,7 @@ func (te *failingExtractor) Extract(filename string, r io.ReadSeeker) (string, e
 }
 
 func TestExtractWithExtraExtractors(t *testing.T) {
-	t.Run("overrite existing extractor", func(t *testing.T) {
+	t.Run("override existing extractor", func(t *testing.T) {
```
Comment (jsoref) [inline services/docextractor/docextractor_test.go:183]: ?

sometimes people want `overwrite`, sometimes `override`, it depends on whether the effect is to stomp on content or change behavior.

File: shared/mail/mail_test.go [hunk 21]
```diff
@@ -226,7 +226,7 @@ func TestSendMailUsingConfigAdvanced(t *testing.T) {
 	}
 
 	err = sendMailUsingConfigAdvanced(mail, cfg)
-	require.NoError(t, err, "Should connect to the STMP Server: %v", err)
+	require.NoError(t, err, "Should connect to the SMTP Server: %v", err)
```
Comment (jsoref) [inline shared/mail/mail_test.go:229]: Protocol

File: store/sqlstore/channel_store.go [hunk 22]
```diff
@@ -1290,7 +1290,7 @@ func (s SqlChannelStore) GetPrivateChannelsForTeam(teamId string, offset int, li
 
 	_, err = s.GetReplica().Select(&channels, query, args...)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to find chaneld with teamId=%s", teamId)
+		return nil, errors.Wrapf(err, "failed to find channel with teamId=%s", teamId)
```
Comment (jsoref) [inline store/sqlstore/channel_store.go:1293]: ?

File: templates/cloud_upgrade_request_email.html [hunk 23]
```diff
@@ -46,7 +46,7 @@ <h2 style="margin-bottom: 0; text-align: center; color: #3D3C40; font-weight: bo
                                 </td>
                             </tr>
                         </table>
-                        <table align="center" border="0" cellpadding="0" cellspcing="0" width="612px">
+                        <table align="center" border="0" cellpadding="0" cellspacing="0" width="612px">
```
Comment (jsoref) [inline templates/cloud_upgrade_request_email.html:49]: behavior change

File: utils/merge_test.go [hunk 24]
```diff
@@ -448,7 +448,7 @@ func TestMergeWithSlices(t *testing.T) {
 		assert.Equal(t, expected, merged)
 	})
 
-	t.Run("patch overwites when patch is empty struct", func(t *testing.T) {
+	t.Run("patch overwrites when patch is empty struct", func(t *testing.T) {
```
Comment (jsoref) [inline utils/merge_test.go:451]: see earlier about override/overwrite

**Review by jsoref — COMMENTED** [review]
Most corrections were suggested by Google Sheets. But all mistakes are mine.

I only saw one (and that was definitely manual).

I'm happy to split things/drop things/etc.

**Comment by agnivade** [comment]
@jsoref - Thanks for working on this. However, I'd have to request you to split this into smaller chunks. As is, it's too big to be reviewed in a single shot.

I'd first suggest to fix only code comments, then strings, and then lastly come to struct fields and others.

**Comment by jsoref** [comment]
@agnivade: No problem.

**Comment by jsoref** [comment]
Oh, sorry, I can drop the strings.

**Comment by jsoref** [comment]
Would it help if i dropped `Makefile`?

**Comment by agnivade** [comment]
That's fine @jsoref. No need to change anything.

File: NOTICE.txt [hunk 25]
```diff
@@ -4494,7 +4494,7 @@ SOFTWARE.
 
 This product contains 'archiver' by Matthew Holt
 
-A library to handle direferen archive files (zip, rar, tar.gz...)
+A library to handle different archive files (zip, rar, tar.gz...)
```
Comment (agnivade) [inline NOTICE.txt:4497]: @amyblais - Is this line written by us or copied from somewhere else?

File: app/server_test.go [hunk 26]
```diff
@@ -548,7 +548,7 @@ func TestPanicLog(t *testing.T) {
 	s, err := NewServer(SetLogger(logger))
 	require.NoError(t, err)
 
-	// Route for just panicing
+	// Route for just panicking
```
Comment (agnivade) [inline app/server_test.go:551]: This is referring to the "panic" feature in the language. Not to be confused with the english word "panic". So let's leave this one as-is, even if it's not a valid word.

File: app/server_test.go [hunk 27]
```diff
@@ -685,7 +685,7 @@ func TestSentry(t *testing.T) {
 		})
 		require.NoError(t, err)
 
-		// Route for just panicing
+		// Route for just panicking
```
Comment (agnivade) [inline app/server_test.go:688]: Same as above

**Review by agnivade — COMMENTED** [review]
Thanks a lot for this @jsoref ! Just some small suggestions. Looks great otherwise

File: NOTICE.txt (hunk 25 above)
Comment (jsoref) [reply NOTICE.txt:4497]: @jespino added it in #15665

Comment (agnivade) [reply NOTICE.txt:4497]: Ah thanks for the blame analysis @jsoref. Good to be changed. :+1:

**Review by agnivade — APPROVED** [review]
Great stuff @jsoref !

**Review by isacikgoz — APPROVED** [review]
LGTM

**Pull request merged**